wav_test.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+type wavExpectedHeader struct {
+	RIFFHeader struct {
+		Ident     string
+		ChunkSize uint32
+		FileType  string
+	}
+	RIFFChunkFmt   riffChunkFmt
+	FirstSamplePos uint32
+	DataBlockSize  uint32
+}
+
+func checkerr(t *testing.T, err error) {
+	t.Helper()
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func testParseWAV(t *testing.T, filename string) {
+	r, err := os.Open(filename)
+	checkerr(t, err)
+
+	hdr, hdrerr := parseHeader(r)
+
+	ext := filepath.Ext(filename)
+	noext := string(filename[0 : len(filename)-len(ext)])
+	expectedHdrFile := noext + ".hdr.expected"
+	errFile := noext + ".err"
+
+	expectedErr, err := ioutil.ReadFile(errFile)
+	if err == nil {
+		if hdrerr == nil {
+			t.Fatalf("Expected error: %s but ran successfully...", string(expectedErr))
+		}
+		if hdrerr.Error() != string(expectedErr) {
+			t.Fatalf("Error differs: '%s' != '%s'", hdrerr, string(expectedErr))
+		}
+		return
+	} else if hdrerr != nil {
+		t.Fatalf("Error: %s", hdrerr)
+	}
+
+	expectedHdrContent, err := ioutil.ReadFile(expectedHdrFile)
+	if err == nil {
+		var expectedHdr wavExpectedHeader
+		err = json.Unmarshal(expectedHdrContent, &expectedHdr)
```

Comment:
A little confused on why use JSON to load the expected HDR. A Go file with expected structs would be too bad ? (if they are too big to include on the test file)

Reply:
How can I load Go structs dinamically ?
I made this way to make simple to add new test cases. This way requires only to drop files in the testdata folder with convention below:

`````
testdata/
    abcde.wav
    abcde.hdr.expected
    xyz.wav
    xyz.err
```

maybe use another format (instead of json), but which one you would recommend?

Reply:
But I need to say that I'm not convinced this is the best way also... Suggestions welcome.

Reply:
Ow I see, well if you are going for dynamic this really seems to be the best way =)